src/googlenet.jl

```diff
@@ -1,82 +1,88 @@
-struct InceptionBlock
-  path_1
-  path_2
-  path_3
-  path_4
-end
-
-@functor InceptionBlock
+"""
+    inceptionblock(inplanes, out_1x1, red_3x3, out_3x3, red_5x5, out_3x3, pool_proj)
 
-function InceptionBlock(in_chs, chs_1x1, chs_3x3_reduce, chs_3x3, chs_5x5_reduce, chs_5x5, pool_proj)
-  path_1 = Conv((1, 1), in_chs=>chs_1x1, relu)
+Create an inception module for use in GoogLeNet
+([reference](https://arxiv.org/abs/1409.4842v1)).
 
-  path_2 = (Conv((1, 1), in_chs=>chs_3x3_reduce, relu),
-            Conv((3, 3), chs_3x3_reduce=>chs_3x3, relu, pad = (1, 1)))
+# Arguments
+- `inplanes`: the number of input feature maps
+- `out_1x1`: the number of output feature maps for the 1x1 convolution (branch 1)
+- `red_3x3`: the number of output feature maps for the 3x3 reduction convolution (branch 2)
+- `out_3x3`: the number of output feature maps for the 3x3 convolution (branch 2)
+- `red_5x5`: the number of output feature maps for the 5x5 reduction convolution (branch 3)
+- `out_5x5`: the number of output feature maps for the 5x5 convolution (branch 3)
+- `pool_proj`: the number of output feature maps for the pooling projection (branch 4)
+"""
+function inceptionblock(inplanes, out_1x1, red_3x3, out_3x3, red_5x5, out_5x5, pool_proj)
```

Comment:
Could definitely use a nicer API here. I am not sure if we should break this down into multiple "specs" for a conv layer, but seems like we want to have access to the filter sizes, as well as manage how the reduction blocks are sized. Currently, its got too many arguments (which would also be the case if these were made kwargs, which I don't think is a good idea.)

we should also remove the `Chain` in the first block.

Reply:
Having objects shouldn't deter from being functional, and it seems like Inception might an example of a  model that would definitely benefit from a struct that holds logical components together.

Reply:
Yeah, I agree it's awful. I would characterize this as an internal function (i.e. maybe we rename it to `_inceptionblock`), because the only reason we have it is to make writing `googlenet` below less verbose. It's absolutely not flexible enough to be used by users. Perhaps I could even define it inside `googlenet` so it stays private.